lm_eval/logging_utils.py

```diff
@@ -398,23 +398,26 @@ def log_eval_samples(self, samples: Dict[str, List[Dict[str, Any]]]) -> None:
 
 
 def get_commit_from_path(repo_path: Path) -> Optional[str]:
-    git_folder = Path(repo_path, ".git")
-    if git_folder.is_file():
-        git_folder = Path(
-            git_folder.parent,
-            git_folder.read_text(encoding="utf-8").split("\n")[0].split(" ")[-1],
-        )
-    if Path(git_folder, "HEAD").exists():
-        head_name = (
-            Path(git_folder, "HEAD")
-            .read_text(encoding="utf-8")
-            .split("\n")[0]
-            .split(" ")[-1]
-        )
-        head_ref = Path(git_folder, head_name)
-        git_hash = head_ref.read_text(encoding="utf-8").replace("\n", "")
-    else:
-        git_hash = None
+    try:
+        git_folder = Path(repo_path, ".git")
+        if git_folder.is_file():
+            git_folder = Path(
+                git_folder.parent,
+                git_folder.read_text(encoding="utf-8").split("\n")[0].split(" ")[-1],
+            )
+        if Path(git_folder, "HEAD").exists():
+            head_name = (
+                Path(git_folder, "HEAD")
+                .read_text(encoding="utf-8")
+                .split("\n")[0]
+                .split(" ")[-1]
+            )
+            head_ref = Path(git_folder, head_name)
+            git_hash = head_ref.read_text(encoding="utf-8").replace("\n", "")
+        else:
+            git_hash = None
+    except Exception:
+        return None
```

Comment:
Adding to the idea from this comment https://github.com/EleutherAI/lm-evaluation-harness/issues/1603#issuecomment-2013646466 to just return None, I suggest to report that something is not good in logging. For example
```
except Exception as err:
    logger.info(f"Logging commit hash failed with error {str(err)}\n Returning None instead")
```

Reply:
or even better in my opinion, to just properly check existence of path
```python
if head_ref.exists():
    git_hash = head_ref.read_text(encoding="utf-8").replace("\n", "")
else:
    git_hash = None
```